C7FIT/ScheduleViewController.swift

```diff
@@ -41,4 +55,149 @@ class ScheduleViewController: UIViewController {
         let rightView = scheduleView.rightAnchor.constraint(equalTo: view.rightAnchor)
         NSLayoutConstraint.activate([topView, bottomView, leftView, rightView])
     }
+    
+    // MARK: UITableView Delegate and Datasource
+    
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        return 1
+    }
+    
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return 3
+    }
+    
+    override func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        let aspectRatioFirst: CGFloat = 1/2
+        let aspectRatioSecond: CGFloat = 3/8
+        let aspectRatioThird: CGFloat = 1/8
+        let screenSize : CGRect = UIScreen.main.bounds
+        //use optionals here
+        //self.navigationController?...
+        let navBarSize : CGFloat? = self.navigationController?.navigationBar.frame.size.height
+        let tabBarSize : CGFloat? = self.tabBarController?.tabBar.frame.size.height
+        let statusBarSize: CGFloat? = UIApplication.shared.statusBarFrame.height
+        let barConstants = screenSize.height - (navBarSize! + tabBarSize! + statusBarSize!)
+        if indexPath.row < 1 {
+            return aspectRatioFirst * barConstants
+        } else if indexPath.row == 1 {
+            return aspectRatioSecond * barConstants //clean this up later if want different
+        } else {
+            return aspectRatioThird * barConstants //clean this up later if want different
+        }
+    }
+    
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        if indexPath.row == 0 {
+            if let cell: ScheduleBrowserTableViewCell = tableView.dequeueReusableCell(withIdentifier: "BrowserLinkCell") as? ScheduleBrowserTableViewCell {
```

Comment:
if you use  dequeueReusableCell(withIdentifier: for:) then you don't get an optional cell returned -- makes for cleaner code